scripts/ecosyst.ms-api.py

```diff
@@ -54,8 +56,24 @@ def processPaper(paperURL):
                     if sdg["id"] not in sdgSet:
                         sdgSet.add(sdg["id"])
                         rowList.append({'ID': sdg['id'], 'Label': "SDG", 'Name': sdg['display_name'], 'sdg_score': sdg['score']})
-
-            rowList.append({'ID': paperDict['openalex_id'], 'Label': "Paper", 'Name': paperDict['title'], 'DOI': paperDict['doi'], 'Authors': " | ".join(paperAuthorNames), 'Projects/Packages Cited': " | ".join(paperMentions), 'Sustainable Development Goals': " | ".join(thisPaperSDGs)})
+                #thisPaperKeywords = []
+                #for key in paperDict['openalex_data']['keywords']:
+                #    thisPaperKeywords.append(key['keyword'])
+                thisPaperConcepts = []
+                for concept in paperDict['openalex_data']['concepts']:
+                    thisPaperConcepts.append(concept['display_name'])
+                    if concept["id"] not in conceptSet:
+                        conceptSet.add(concept["id"])
+                        rowList.append({'ID': concept['id'], 'Label': "Concept", 'Name': concept['display_name'], 'Wikidata': concept['wikidata'], 'Concept_level': concept['level'] })
+                thisPaperDomains = []
+                for domain in paperDict['openalex_data']['mesh']:
+                    thisPaperDomains.append(domain['descriptor_name'])
+                    if domain["descriptor_ui"] not in conceptSet:
+                        domainSet.add(domain["descriptor_ui"])
+                        rowList.append({'ID': domain['descriptor_ui'], 'Label': "Domain", 'Name': domain['descriptor_name'], 'Is_major_topic': domain['is_major_topic'] })
```

Comment:
Consider adding line comments to help summarize what's happening at reasonable points. Code comments could help to illuminate or confirm what is happening here (the code shows logically what's happening but might be difficult to reason through without guidance).

Reply:
I've added some line comments, plan to be more verbose after a refactor next PR